src/linux_mcp_server/utils/format.py

```diff
@@ -1,3 +1,41 @@
+import ipaddress
+
+
+def is_ipv6_link_local(address: str) -> bool:
+    """Check if an IPv6 address is link-local (fe80::/10).
+
+    Link-local addresses are only valid on a single network segment and
+    require a scope identifier to route. They're not useful for enterprise
+    cross-machine communication.
+
+    Args:
+        address: IPv6 address string, possibly with scope identifier (e.g., "fe80::1%eth0")
+
+    Returns:
+        True if the address is link-local, False otherwise
+
+    Examples:
+        >>> is_ipv6_link_local("fe80::1")
+        True
+        >>> is_ipv6_link_local("fe80::1%eth0")
+        True
+        >>> is_ipv6_link_local("2001:db8::1")
+        False
+        >>> is_ipv6_link_local("192.168.1.1")
+        False
+    """
+    try:
+        # Strip scope identifier (e.g., %eth0) if present
+        addr_str = address.split("%")[0]
```

Comment:
It's not necessary to remove the scope zone ID. It's a [valid part](https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv6Address) of the address.

Reply:
Ah good catch. I thought the ipaddress module would explode with that, but it doesn't!